src/porepy/fracs/fractures_2d.py

```diff
@@ -232,16 +235,26 @@ def mesh(
             # Delete constraints corresponding to deleted edges
             constraints = np.delete(constraints, to_delete)
 
+        # Consider the dimension of the problem, normally 2d but if dfn it's a 1d
+        ndim = 2-int(dfn)
+
         self._find_and_split_intersections(constraints)
         self._insert_auxiliary_points(**mesh_args)
-        self._to_gmsh(in_file)
+        self._to_gmsh(in_file, ndim=ndim)
+        pp.grids.gmsh.gmsh_interface.run_gmsh(in_file, out_file, dim=ndim)
 
-        pp.grids.gmsh.gmsh_interface.run_gmsh(in_file, out_file, dim=2)
+        if dfn:
+            # Create list of grids
+            grid_list = porepy.fracs.simplex.line_grid_from_gmsh(
```

Comment:
I guess it is okay to keep the constraints for consistency, but I don't see how it can be meaninful to have constraints in a dfn grid.